share/h2o/mruby/bootstrap.rb

```diff
@@ -1,16 +1,24 @@
 module H2O
 
-  @@handler_validators = []
+  @@hooks = {
+    :after_generate_handler => [],
+  }
 
-  def self.validate_handler(handler)
-    @@handler_validators.each do |validator|
-      # will raise exception if handler is not valid
-      validator.call(handler)
-    end
+  def self.after_generate_handler(handler)
+    _call_hooks(:after_generate_handler, handler)
   end
 
-  def self.add_handler_validator(validator)
-    @@handler_validators << validator
+  def self.add_after_generate_handler_hook(hook)
+    _add_hook(:after_generate_handler, hook)
+  end
+
+  def self._call_hooks(type, *args)
+    @@hooks[type].each {|hook| hook.call(*args) }
+    @@hooks[type].clear
+  end
+
+  def self._add_hook(type, hook)
+    @@hooks[type] << hook
```

Comment:
@kazuho To get mruby state shared between multiple handler contexts, each handler context occasionally should do some `clean up` or `finalize` process, additional to validation. For example, `acl` module has a state which is used to validating handler, and that state should be used only in the scope of one handler configuration. So it should be cleared after the handler context is generated.
So, we should have two options:
1. Adding another hook, like `cleanup`, and call it in `on_context_init` too
2. Put validation and cleanup together in opaque `hook` s

I felt that the latter is better because of ... simplicity 😄 (It's difficult for me to explain this in English, so let's talk f2f later)